### PR TITLE
Reconstruct and reorganize the model_demos directory

### LINKED ISSUE #769: Reconstruct and reorganize  model_demos
### Current directory structure:
```
model_demos/
├── __init__.py
├── high_prio
│   ├── cnn
│   │   ├── __init__.py
│   │   ├── onnx
│   │   │   ├── __init__.py
│   │   │   ├── test_model_1.py
│   │   │   ├── ...
│   │   │   ├── ...
│   │   │   └── test_model_n.py
│   │   ├── pytorch
│   │   │   ├── __init__.py
│   │   │   ├── test_model_1.py
│   │   │   ├── ...
│   │   │   ├── ...
│   │   │   └── test_model_n.py
│   │   └── tflite
│   │       ├── __init__.py
│   │       ├── test_model_1.py
│   │       ├── ...
│   │       ├── ...
│   │       └── test_model_n.py
│   └── nlp
│       ├── __init__.py
│       └── pytorch
│           ├── __init__.py
│           ├── test_model_1.py
│           ├── ...
│           ├── ...
│           ├── test_model_n.py
├── models
│   ├── __init__.py
│   ├── model_script_1.py
│   ├── ...
│   ├── ...
│   ├── model_script_n.py
└── utils
    ├── __init__.py
    ├── cnn
    │   ├── onnx
    │   │   └── images
    │   │       └── carvana.jpg
    │   └── pytorch
    │       ├── images
    │       │   ├── car.jpg
    │       │   ├── girl.png
    │       │   └── img.jpeg
    │       └── saved
    │           ├── efficientnet_lite
    │           │   └── src_efficientnet_lite.py
    │           ├── mobilenetv1_ssd
    │           │   └── vision
    │           │       └── nn
    │           │           └── mobilenet.py
    │           └── yolo_v3
    │               └── holli_src
    │                   └── license
    └── nlp
        └── pytorch
            ├── 1272-128104-0000.pt
            ├── 1272-128104-0001.pt
            ├── 1272-128104-0002.pt
            └── 1272-128104-0003.pt
```

### New directory structure:
```
models/
├── __init__.py
├── files/
│   ├── samples/
│       ├── cat.png
│       └── dog.png
└── framework_name/
    ├── __init__.py
    ├── vision/
    │   ├── __init__.py
    │   ├── resnet/
    │   │   ├── __init__.py
    │   │   ├── test_resnet_inference.py
    │   │   ├── test_resnet_training.py
    │   │   ├── tests/ (Optional)
    │   │   │   ├── __init__.py
    │   │   │   ├── blocks/
    |   |   |   |    ├── __init__.py
    │   │   │   |    └── test_block1.py
    |   │   │   |    └── test_block2.py
    |   │   │   |    └── test_blockn.py
    │   │   │   └── Ops/
    |   |   |   |    ├── __init__.py
    │   │   │   |    └── test_ops.py
    │   │   └── utils/
    │   │       ├── __init__.py
    │   │       └── resnet.py
    │   └── yolo/
    │       ├── __init__.py
    │       ├── test_yolox_inference.py
    │       ├── test_yolox_training.py
    │       ├── tests/ (Optional)
    │       │   ├── __init__.py
    │       │   ├── blocks/
    |       |   |    ├── __init__.py
    │       │   |    └── test_block1.py
    |       │   |    └── test_block2.py
    |       │   |    └── test_blockn.py
    │       │   └── ops/
    |       |   |    ├── __init__.py
    │       │   |    └── test_ops.py
    │       └── utils/
    │           ├── __init__.py
    │           └── yolox.py
    ├── text/
        ├── __init__.py
        ├── llama/
            ├── __init__.py
            ├──test_llama_inference.py
            ├──test_llama_training.py
            ├── tests/ (Optional)
            │   ├── __init__.py
            │   ├── blocks/
            |   |    ├── __init__.py
            │   |    └── test_block1.py
            │   |    └── test_block2.py
            │   |    └── test_blockn.py
            │   └── ops/
            |   |    ├── __init__.py
            │   |    └── test_ops.py
            └── utils/
                ├── __init__.py
                └── llama.py
```

- **`models/`**: Root directory for framework-specific subdirectories (e.g., PyTorch, ONNX, TensorFlow)

  - **`files/`**: Directory contains models input samples, weight and onnx files

  - **`framework_name/`**: Directory for a specific framework (e.g., PyTorch, TensorFlow, ONNX)

    - **`vision/`**: Directory contains models designed to process image data as input. 

      - **`Model_Name/`**: Directory contains model specific details like model test cases like inference, training, ops/block specific test and utils directory contain model implementation 

    - **`text/`**: Directory contains models designed to process text data as input. 

      - **`Model_Name/`**:   Directory contains model specific details like model test cases like inference, training, ops/block specific test and utils directory contain model implementation

### PR DESCRIPTION
**Pull request by chandrasekaranpradeep**
The PR will Reconstruct and reorganize the models demos directory and added script for creating  package for new models based upon the model name, framework and model category 
Fixes #769

### CODE REVIEW

File: forge/test/create_model_structure.sh [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+#
+# SPDX-License-Identifier: Apache-2.0
+
+# !/bin/bash
+
+# : << 'COMMENT'
+# SAMPLE:
+# Script run command : bash forge/test/create_model_structure.sh
+
+# INPUTS:
```
Comment (nvukobratTT) [inline forge/test/create_model_structure.sh:11]: Let's create a doc section for this script as well.

**Review by nvukobratTT — APPROVED** [review]
@dgolubovicTT once this change is merge, we should think about moving our core models to this structure :))

Nothing urgent, just something to think about as there are some of the models that overlap.

**Comment by nvukobratTT** [comment]
@chandrasekaranpradeep let's update "Category" section of our model tracking document to math this (vision, text, etc.)

File: forge/test/create_model_structure.sh (hunk 1 above)
Comment (chandrasekaranpradeep) [reply forge/test/create_model_structure.sh:11]: Sure Let me create a issue for adding documenting the script.

Comment (chandrasekaranpradeep) [reply forge/test/create_model_structure.sh:11]: Created a issue for the documenting the script - https://github.com/tenstorrent/tt-forge-fe/issues/801

**Pull request merged**